toolkit/tools/imagegen/configuration/fileconfig.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//
+
+package configuration
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// DestinationFileConfigList is a list of destination files to copy.
+type FileConfigList []FileConfig
```

Comment:
We haven't used types to mask lists before, but I'm not opposed to it if you feel it makes things cleaner.

Reply:
This type only exists to allow me to attach a custom marshaller to it.